File: app/src/main/java/workshop/akbolatss/tagsnews/model/ReminderItemRepository.java
package workshop.akbolatss.tagsnews.model;


import java.util.List;

import io.reactivex.Single;
import workshop.akbolatss.tagsnews.model.dao.ReminderItem;

public interface ReminderItemRepository {

    Single<List<ReminderItem>> onLoadReminders();

    void onAddReminder(ReminderItem rItem);

    void onRemoveReminder(ReminderItem rItem);

    void onUpdateReminder(ReminderItem rItem);

    void onActivateReminder(ReminderItem rItem);

    void onDeactivateReminder(ReminderItem rItem);
}

File: app/src/main/java/workshop/akbolatss/tagsnews/model/ReminderItemRepository.kt
package workshop.akbolatss.tagsnews.model


import io.reactivex.Single
import workshop.akbolatss.tagsnews.model.dao.ReminderItem

/**
 * @see DBReminderItemRepository
 */
interface ReminderItemRepository {

    fun loadReminders(): Single<List<ReminderItem>>

    fun addReminder(rItem: ReminderItem)

    fun removeReminder(rItem: ReminderItem)

    fun updateReminder(rItem: ReminderItem)

    fun activateReminder(rItem: ReminderItem)

    fun deactivateReminder(rItem: ReminderItem)
}
